### PR TITLE
[fix]: [DEL-0]: Fix stuck delegate tasks on agent

### PR DESCRIPTION
**Pull request by shubham149**
You can use the following comments to re-trigger PR Checks

- Compile: `trigger compile`
- runAeriformCheck: `trigger AeriformCheck`
- CodeFormat: `trigger codeformat`
- MessageMetadata: `trigger messagecheck`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- runDockerizationCheck: `trigger dockerizationcheck`
- runAuthorCheck: `trigger authorcheck`
- Checkstyle: `trigger checkstyle`
- PMD: `trigger pmd`
- TI-bootstrap: `trigger ti0`
- TI-bootstrap1: `trigger ti1`
- TI-bootstrap2: `trigger ti2`
- TI-bootstrap3: `trigger ti3`
- TI-bootstrap4: `trigger ti4`
- FunctionalTest1: `trigger ft1`
- FunctionalTest2: `trigger ft2`
- CodeBaseHash: `trigger codebasehash`

---
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/30438)

### CODE REVIEW

**Review by abhinav-harness — APPROVED** [review]
> Reviewed 1 of 1 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/30438)* status: :shipit: complete! ✓

**Comment by shubham149** [comment]
trigger messagecheck

File: 260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java [hunk 1]
```diff
@@ -2062,6 +2062,7 @@ private void executeTask(@NotNull DelegateTaskPackage delegateTaskPackage) {
     timeoutEnforcement.submit(() -> enforceDelegateTaskTimeout(delegateTaskPackage.getDelegateTaskId(), taskData));
 
     // Start task execution in same thread and measure duration.
+    delegateRunnableTask.run();
```
Comment (gaurav-jain1) [inline 260-delegate/src/main/java/io/harness/delegate/service/DelegateAgentServiceImpl.java:2065]: Should recordGaugeDuration call happen before this run call ? @bojanmicovic

**Comment by shubham149** [comment]
trigger compile

**Comment by abhinav-harness** [comment]
@shubham149 we have created a new PR closing this one
New PR - https://github.com/harness/harness-core/pull/30438
cc: @bojanmicovic

**Pull request closed without merging**